### PR TITLE
Salto 4599 deploy

### PR DESCRIPTION
**Pull request by idoser96**
_Replace me with a description of the changes in this PR_

---

_Additional context for reviewer_

---
_Release Notes_: 
_Replace me with a short sentence that describes the effect of this change on Salto users_

---
_User Notifications_: 
_Replace me with a short sentence that describes changes that will appear in NaCls and are not caused by user actions (e.g. a new annotation, field values that are converted to references, etc). Hidden changes should not be listed._

### CODE REVIEW

File: packages/zendesk-adapter/src/filters/custom_objects/custom_object_fields.ts [hunk 1]
```diff
@@ -250,59 +270,149 @@ const transformTicketAndCustomObjectFieldValue = ({
...
+  const ticketFields = instances.filter(instance => instance.elemID.typeName === TICKET_FIELD_TYPE_NAME)
+  const customObjectFields = instances.filter(instance => instance.elemID.typeName === CUSTOM_OBJECT_FIELD_TYPE_NAME)
+
+  const triggerConditions = triggers.flatMap(trigger => filterUserConditions(
+    trigger.value.conditions,
+    trigger.elemID.getFullNameParts().concat('conditions'),
+    isRelevantCondition
+  ))
+
+  const ticketAndCustomObjectFieldFilters = ticketFields.concat(customObjectFields).flatMap(field =>
```
Comment (idoser96) [inline packages/zendesk-adapter/src/filters/custom_objects/custom_object_fields.ts]: This seem possible, but i did not manage to create it in the UI

File: packages/zendesk-adapter/src/config.ts [hunk 2]
```diff
@@ -2513,6 +2527,9 @@ export const DEFAULT_TYPES: ZendeskApiConfig['types'] = {
       fieldsToHide: FIELDS_TO_HIDE.concat(
         { fieldName: 'created_by_user_id', fieldType: 'string' },
         { fieldName: 'updated_by_user_id', fieldType: 'string' },
+        { fieldName: 'title', fieldType: 'string' },
+        { fieldName: 'title_pluralized', fieldType: 'string' },
+        { fieldName: 'description', fieldType: 'string' },
```
Comment (edenhassid) [inline packages/zendesk-adapter/src/config.ts:2535]: why are you hiding these fields?

File: packages/zendesk-adapter/src/config.ts [hunk 3]
```diff
@@ -2564,6 +2581,10 @@ export const DEFAULT_TYPES: ZendeskApiConfig['types'] = {
       fieldsToHide: FIELDS_TO_HIDE.concat(
         { fieldName: 'id', fieldType: 'number' },
       ),
+      fieldsToOmit: FIELDS_TO_OMIT.concat(
+        { fieldName: 'description', fieldType: 'string' },
+        { fieldName: 'title', fieldType: 'string' },
```
Comment (edenhassid) [inline packages/zendesk-adapter/src/config.ts]: why do you omit these fields?

File: packages/zendesk-adapter/src/filters/custom_objects/custom_object_fields.ts [hunk 4]
```diff
@@ -267,59 +282,139 @@ const transformTicketAndCustomObjectFieldValue = ({
...
+const getUserConditions = (changes: Change<InstanceElement>[]): CustomObjectCondition[] => {
+  const instances = changes
+    .filter(isAdditionOrModificationChange)
+    .map(getChangeData)
+
+  const triggers = instances.filter(instance => instance.elemID.typeName === TRIGGER_TYPE_NAME)
+  const ticketFields = instances.filter(instance => instance.elemID.typeName === TICKET_FIELD_TYPE_NAME)
+  const customObjectFields = instances.filter(instance => instance.elemID.typeName === CUSTOM_OBJECT_FIELD_TYPE_NAME)
+
+  const triggerConditions = triggers.flatMap(trigger =>
```
Comment (edenhassid) [inline packages/zendesk-adapter/src/filters/custom_objects/custom_object_fields.ts]:
```suggestion
  const triggerUserConditions = triggers.flatMap(trigger =>
```

File: packages/zendesk-adapter/src/filters/custom_objects/custom_object_fields.ts [hunk 5]
```diff
@@ -267,59 +282,139 @@ const transformTicketAndCustomObjectFieldValue = ({
...
+    filterUserConditions(trigger.value.conditions, isRelevantCondition))
+
+  const ticketAndCustomObjectFieldFilters = ticketFields.concat(customObjectFields).flatMap(field =>
```
Comment (edenhassid) [inline packages/zendesk-adapter/src/filters/custom_objects/custom_object_fields.ts]:
```suggestion
  const ticketAndCustomObjectFieldUserUserFilter = ticketFields.concat(customObjectFields).flatMap(field =>
```

File: packages/zendesk-adapter/src/filters/custom_objects/custom_object_fields.ts [hunk 6]
```diff
@@ -267,59 +282,139 @@ const transformTicketAndCustomObjectFieldValue = ({
...
 
+      const { defaultMissingUserFallback } = config[DEPLOY_CONFIG] ?? {}
+      if (missingUserConditions.length > 0 && defaultMissingUserFallback !== undefined) {
+        const userEmails = new Set(users.map(user => user.email))
+        const fallbackValue = await getUserFallbackValue(
+          defaultMissingUserFallback,
+          userEmails,
+          client
+        )
+        if (fallbackValue !== undefined && usersByEmail[fallbackValue]) {
```
Comment (edenhassid) [inline packages/zendesk-adapter/src/filters/custom_objects/custom_object_fields.ts]:
```suggestion
        if (fallbackValue !== undefined && usersByEmail[fallbackValue] !== undefined) {
```

File: packages/zendesk-adapter/src/filters/custom_objects/custom_object_fields.ts [hunk 7]
```diff
@@ -267,59 +282,139 @@ const transformTicketAndCustomObjectFieldValue = ({
...
+            condition.value = fallbackUserId
+          })
+        } else {
+          log.error('Error while trying to get defaultMissingUserFallback value in customObjectFieldsFilter')
+        }
+      }
+    },
+    onDeploy: async (changes: Change<InstanceElement>[]) => {
+      getUserConditions(changes).forEach(condition => {
+        condition.value = _.isString(condition.value) && userPathToOriginalValue[condition.value]
```
Comment (edenhassid) [inline packages/zendesk-adapter/src/filters/custom_objects/custom_object_fields.ts]:
```suggestion
        condition.value = _.isString(condition.value) && userPathToOriginalValue[condition.value] !== undefined
```

**Review by edenhassid — COMMENTED** [review]
looks good! added a few comments

File: packages/zendesk-adapter/src/config.ts (hunk 2 above)
Comment (idoser96) [reply packages/zendesk-adapter/src/config.ts:2535]: Moved to omit and added a comment

File: packages/zendesk-adapter/src/config.ts (hunk 3 above)
Comment (idoser96) [reply packages/zendesk-adapter/src/config.ts]: Added a comment

**Review by edenhassid — APPROVED** [review]
LGTM!

**Pull request merged**